File: build-projects.js
const fs = require('fs');
const path = require('path');

const categories = { '3d': '3D', 'design': 'Design', 'it': 'IT' };
const baseDir = path.join(__dirname, 'Projects');
const fallbackData = {};

Object.entries(categories).forEach(([key, folderName]) => {
    const targetDir = path.join(baseDir, folderName);
    let projects = [];

    if (fs.existsSync(targetDir)) {
        const categoryDisabled = fs.existsSync(path.join(targetDir, 'OFF'));
        const items = fs.readdirSync(targetDir, { encoding: 'utf8' });
        for (const item of items) {
            const itemPath = path.join(targetDir, item);
            if (fs.statSync(itemPath).isDirectory()) {
                let files = fs.readdirSync(itemPath, { encoding: 'utf8' }).filter(f => {
                    const ext = path.extname(f).toLowerCase();
                    return ['.jpg', '.jpeg', '.png', '.webp', '.pdf', '.mp4', '.webm'].includes(ext);
                });

                // Sort slides by number prefix if available
                files.sort((a, b) => {
                    const numA = parseInt(a.match(/^\d+/)) || 9999;
                    const numB = parseInt(b.match(/^\d+/)) || 9999;
                    return numA - numB || a.localeCompare(b);
                });

                if (files.length > 0) {
                    const isProjectDisabled = categoryDisabled || fs.existsSync(path.join(itemPath, 'OFF'));
                    projects.push({ folder: item, files, disabled: isProjectDisabled });
                }
            }
        }
    }

    // Sort projects in inverse order (highest number prefix first) — matches PHP API
    projects.sort((a, b) => {
        const numA = parseInt(a.folder.match(/^\d+/)) || -1;
        const numB = parseInt(b.folder.match(/^\d+/)) || -1;
        if (numA !== numB) return numB - numA;
        return a.folder.localeCompare(b.folder);
    });
    fallbackData[key] = projects;
});

// Write to api folder
const outPath = path.join(__dirname, 'api', 'projects-fallback.json');
fs.writeFileSync(outPath, JSON.stringify(fallbackData, null, 2), 'utf-8');
console.log('✅ Generated fallback projects JSON for local static testing.');

File: api/build-projects.py
#!/usr/bin/env python3
"""Generate projects-fallback.json by scanning the Projects directory."""
import os, json

BASE = os.path.join(os.path.dirname(os.path.abspath(__file__)), '..', 'Projects')
CATS = {'3d': '3D', 'design': 'Design', 'it': 'IT'}
EXTS = {'.jpg', '.jpeg', '.png', '.webp', '.pdf', '.mp4', '.webm'}
data = {}

for key, folder in CATS.items():
    target = os.path.join(BASE, folder)
    projects = []
    if os.path.isdir(target):
        cat_disabled = os.path.isfile(os.path.join(target, 'OFF'))
        for item in sorted(os.listdir(target)):
            item_path = os.path.join(target, item)
            if os.path.isdir(item_path):
                files = sorted([f for f in os.listdir(item_path)
                               if os.path.splitext(f)[1].lower() in EXTS])
                if files:
                    is_disabled = cat_disabled or os.path.isfile(os.path.join(item_path, 'OFF'))
                    projects.append({'folder': item, 'files': files, 'disabled': is_disabled})
    data[key] = projects

out = os.path.join(os.path.dirname(os.path.abspath(__file__)), 'projects-fallback.json')
with open(out, 'w', encoding='utf-8') as f:
    json.dump(data, f, ensure_ascii=False, indent=2)
print(json.dumps(data, ensure_ascii=False, indent=2))
print(f'\nSaved to {out}')
